ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -0,0 +1,301 @@
+        new CachingBlockHashLookup(blockSelectionContext.processableBlockHeader(), blockchain);
+
+    final TransactionProcessingResult effectiveResult =
+        transactionProcessor.processTransaction(
+            blockchain,
+            worldStateUpdater,
+            blockSelectionContext.processableBlockHeader(),
+            transaction,
+            blockSelectionContext.miningBeneficiary(),
+            blockHashLookup,
+            false,
+            TransactionValidationParams.mining(),
+            blockSelectionContext.blobGasPrice());
+
+    var transactionWithProcessingContextResult =
+        evaluateTransaction(transaction, blockSelectionContext, effectiveResult);
+    if (!transactionWithProcessingContextResult.selected()) {
+      return transactionWithProcessingContextResult;
+    }
+
+    final long gasUsedByTransaction = transaction.getGasLimit() - effectiveResult.getGasRemaining();
+    final long cumulativeGasUsed =
+        transactionSelectionResults.getCumulativeGasUsed() + gasUsedByTransaction;
+
+    worldStateUpdater.commit();
+    final TransactionReceipt receipt =
+        transactionReceiptFactory.create(
+            transaction.getType(), effectiveResult, worldState, cumulativeGasUsed);
+
+    final long blobGasUsed =
+        blockSelectionContext.gasCalculator().blobGasCost(transaction.getBlobCount());
+
+    transactionSelectionResults.updateSelected(
+        transaction, receipt, gasUsedByTransaction, blobGasUsed);
+
+    LOG.atTrace()
+        .setMessage("Selected {} for block creation")
+        .addArgument(transaction::toTraceLog)
+        .log();
+
+    return TransactionSelectionResult.SELECTED;
+  }
+
+  public TransactionSelectionResult evaluateTransaction(
+      final Transaction transaction, final BlockSelectionContext context) {
+
+    // Create transaction selectors
+    var selectors = blockTransactionSelectorFactory.createTransactionSelectors(context);
+    var result = evaluateTransactionWithSelectors(transaction, selectors, null);
+    if (!result.selected()) {
+      return result;
+    }
+
+    // Create the plugin selectors
+    var externalSelectors = blockTransactionSelectorFactory.createPluginTransactionSelectors();
+    var externalResult = evaluateTransactionWithSelectors(transaction, externalSelectors);
+    if (!externalResult.selected()) {
+      return externalResult;
+    }
+    // If the transaction is selected by all selectors, return SELECTED
+    return TransactionSelectionResult.SELECTED;
+  }
+
+  public TransactionSelectionResult evaluateTransaction(
+      final Transaction transaction,
+      final BlockSelectionContext context,
+      final TransactionProcessingResult effectiveResult) {
+    List<AbstractTransactionSelector> selectors =
+        blockTransactionSelectorFactory.createTransactionSelectors(context);
+    var result = evaluateTransactionWithSelectors(transaction, selectors, effectiveResult);
+    if (!result.selected()) {
+      return result;
+    }
```

Comment:
I have added a todo, it will be addressed in a next PR.    

// TODO: External selectors are not used here because TransactionProcessingResult is not 
//  exposed to the Plugin API yet.